### ISSUE TITLE
Collecting GC and event loop lag. Could there be any performance issue related?

### ISSUE THREAD

**Issue by Nikoms** [prometheus/client_js#490]
Hello everyone :) And first, thank you for this really good lib!

I have a question about an upgrade we just made...

We just moved from v11 to v14 to be able to collect metrics of GC and eventloop lag. In our grafana charts, we saw a little jump in our API response time (+10ms) and some other things like the `nodejs_heap_space_size_used_bytes` and `nodejs_external_memory_bytes` being less flat than before.

Is it possible that collecting these 2 set of metrics comes also with a little overhead?

Thank you :)

**Comment by GLStephen** [comment]
We've noticed additional event loop lag and some other increasing metrics over time since using prom-client. What confounds this is that we didn't have as clear visibility into the app before we started using it, so we're comparing apples to oranges a bit, but I can confirm a suspicion about the same thing.

**Comment by marwankhalili** [comment]
The gc metrics use PerformanceObserver that is subscribed indefinitely, which could explain the performance overhead (somewhat related to #500).

See Node docs https://nodejs.org/api/perf_hooks.html#class-perf_hooksperformanceobserver

> Because PerformanceObserver instances introduce their own additional performance overhead, instances should not be left subscribed to notifications indefinitely. Users should disconnect observers as soon as they are no longer needed.

**Comment by jdmarshall** [comment]
The largest object in the Event Loop Lag by far is an ELDHistogram, retained by the collect() function. However this change landed shortly before this bug was reported:

https://github.com/siimon/prom-client/pull/459

So I'm wondering if this problem was in fact fixed but not released until later?

**Comment by jdmarshall** [comment]
Please look out for the new version of prometheus client we are about to release. This should lower the slope of memory usage caused by stat cardinality creep.
